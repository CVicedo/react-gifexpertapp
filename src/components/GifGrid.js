import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import PropTypes from 'prop-types'



export const GifGrid = ({category}) => {

    const {data: images, loading} = useFetchGifs( category )

    const imagesYes = (images) => {
        return(
            <div className="card-grid">
                    {images.map((img) => {
                        return(
                            <GifGridItem
                                key={img.id}
                                {...img}
                            />
                        )
                    })}
            </div>
        )
    }

    const imagesNo = () => <p>Sorry, no gifs found.</p>;
        
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            
            {images.length === 0 ? imagesNo() : imagesYes(images) }
            
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
