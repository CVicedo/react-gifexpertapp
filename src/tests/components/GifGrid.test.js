import React from 'react'
import { shallow } from "enzyme"
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en <GifGrid />', () => {
    const category = 'A category'
    
    test('debe mostrarse el componente correctamente', ()=> {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar items cuando se cargan imÃ¡genes con useFetchGifs', () => {
        
        const gifs = [{
            id: 'Gif Id',
            url: 'https://localhost.4000/img.gif',
            title: 'Gif Title'
        },
        {
            id: '123456',
            url: 'https://localhost.4000/img.gif',
            title: 'Gif Title'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={category} />)

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists()).toBeFalsy()
        expect( wrapper.find('GifGridItem').length).toBe(gifs.length)

    })
    
})