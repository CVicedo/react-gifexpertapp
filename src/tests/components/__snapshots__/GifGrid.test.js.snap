// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    A category
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="Gif Id"
      key="Gif Id"
      title="Gif Title"
      url="https://localhost.4000/img.gif"
    />
    <GifGridItem
      id="123456"
      key="123456"
      title="Gif Title"
      url="https://localhost.4000/img.gif"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrarse el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    A category
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <p>
    Sorry, no gifs found.
  </p>
</Fragment>
`;
